dask_expr/tests/test_distributed.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import pytest
+
+distributed = pytest.importorskip("distributed")
+
+import asyncio
+import os
+
+from dask.distributed import Worker
+from distributed.scheduler import Scheduler
+from distributed.utils import Deadline
+from distributed.utils_test import client as c  # noqa F401
+from distributed.utils_test import gen_cluster
+
+import dask_expr as dx
+
+
+async def clean_worker(
+    worker: Worker, interval: float = 0.01, timeout: int | None = None
+) -> None:
+    """Assert that the worker has no shuffle state"""
+    deadline = Deadline.after(timeout)
+    extension = worker.extensions["shuffle"]
+
+    while extension._runs and not deadline.expired:
+        await asyncio.sleep(interval)
+    for dirpath, dirnames, filenames in os.walk(worker.local_directory):
+        assert "shuffle" not in dirpath
+        for fn in dirnames + filenames:
+            assert "shuffle" not in fn
+
+
+async def clean_scheduler(
+    scheduler: Scheduler, interval: float = 0.01, timeout: int | None = None
+) -> None:
+    """Assert that the scheduler has no shuffle state"""
+    deadline = Deadline.after(timeout)
+    extension = scheduler.extensions["shuffle"]
+    while extension.states and not deadline.expired:
+        await asyncio.sleep(interval)
+    assert not extension.states
+    assert not extension.heartbeats
```

Comment:
I strongly recommend using `distributed.utils_tests.gen_cluster`

Reply:
Oh wait, I see that we are.  Never mind, I was confused.  I realize now that this is for shuffling code.  Maybe not a big deal here?  This is already tested in dask/dask?

Reply:
Right, I copied this from the shuffle tests in distributed and thought we may also want to make sure we are not leaving behind any shuffle state when our test finishes. Do you think we can just rely on `gen_cluster` and skip the `clean_worker` and `clean_scheduler` calls?

Reply:
They do very different things.  gen_cluster sets up a Dask cluster.  clean_worker/scheduler verify that no shuffling state leaked out for p2p shuffling.  I think that the p2p shuffling is written and tested in dask/distributed.  I don't think that we need to also test that it functions well in this code base.  Instead, I think we need to verify that it runs, and that things like optimizations work well.

Reply:
I think I'll assume "yes" for now, so we can keep `test_distributed.py` light.

Reply:
I agree that the leakage is already tested for in dask/distributed and doesn't need to be tested here. I'm pretty sure I just misunderstood exactly what those utilities were doing.

I think this PR is now in good shape for initial p2p support.